src/ripple/app/tx/impl/PayChan.cpp

```diff
@@ -220,31 +234,50 @@ PayChanCreate::doApply()
     // Create PayChan in ledger
     auto const slep = std::make_shared<SLE> (
         keylet::payChan (account, dst, (*sle)[sfSequence] - 1));
-    // Funds held in this channel
-    (*slep)[sfAmount] = ctx_.tx[sfAmount];
-    // Amount channel has already paid
-    (*slep)[sfBalance] = ctx_.tx[sfAmount].zeroed ();
-    (*slep)[sfAccount] = account;
-    (*slep)[sfDestination] = dst;
-    (*slep)[sfSettleDelay] = ctx_.tx[sfSettleDelay];
-    (*slep)[sfPublicKey] = ctx_.tx[sfPublicKey];
-    (*slep)[~sfCancelAfter] = ctx_.tx[~sfCancelAfter];
-    (*slep)[~sfSourceTag] = ctx_.tx[~sfSourceTag];
-    (*slep)[~sfDestinationTag] = ctx_.tx[~sfDestinationTag];
-
-    ctx_.view ().insert (slep);
 
-    // Add PayChan to owner directory
+    STArray chanMembers {sfChannelMembers, 2};
+    chanMembers.push_back (STObject (sfChannelMember));
+    STObject& member1 = chanMembers.back ();
+    member1[sfBalance] = ctx_.tx[sfAmount];
+    member1[sfAccount] = account;
+    member1[sfPublicKey] = ctx_.tx[sfPublicKey];
+    member1[sfAmount] = XRPAmount{beast::zero};
+    member1[sfSequence] = 0;
+
+    chanMembers.push_back (STObject (sfChannelMember));
+    STObject& member2 = chanMembers.back ();
+    member2[sfBalance] = XRPAmount{beast::zero};
+    member2[sfAccount] = dst;
+    member2[sfPublicKey] = ctx_.tx[sfDstPublicKey];
+    member2[sfAmount] = XRPAmount{beast::zero};
+    member2[sfSequence] = 0;
+
+    // Add PayChan to owner directories
     {
         uint64_t page;
         auto result = dirAdd (ctx_.view (), page, keylet::ownerDir (account),
             slep->key (), describeOwnerDir (account),
             ctx_.app.journal ("View"));
         if (!isTesSuccess (result.first))
             return result.first;
-        (*slep)[sfOwnerNode] = page;
+        member1[sfOwnerNode] = page;
+
+        auto const sled = ctx_.view ().peek (keylet::account (dst));
+        result = dirAdd (ctx_.view (), page, keylet::ownerDir (dst),
```

Comment:
We add to the owner directory without incrementing the owner count. This behavior makes sense (allowing one account to increase the reserve on another account would be bad). Do we do this anywhere else in rippled (add to owner dir without incrementing count). Is it going to invalidate assumptions if we do it here?